src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -324,6 +324,20 @@ def getTask(self, taskName):
             return None
         return WMTaskHelper(task)
 
+    def getTaskByName(self, taskName):
+        """
+        _getTaskByName_
+
+        Retrieve a task with the given name in the whole workflow tree.
+        """
+        for t in self.taskIterator():
+            if t.name() == taskName:
+                return t
+            for x in t.taskIterator():
```

Comment:
Yes. But if is the top level task, it doesn't have to construct the tree at all when we check it first. So although it is redundant check, we might keep it. But if you are against it we can remove it as well.